finality-aleph/src/validator_network/manager/mod.rs

```diff
@@ -30,191 +41,165 @@ impl Display for SendError {
-            let peers = self
-                .both_ways_peers
-                .iter()
-                .map(|authority_id| authority_id.to_short_string())
-                .collect::<Vec<_>>()
-                .join(", ");
-            write!(
-                f,
-                "both ways - {:?} [{}]; ",
-                self.both_ways_peers.len(),
-                peers,
-            )?;
-        }
-
-        if !self.incoming_peers.is_empty() {
-            let peers = self
-                .incoming_peers
-                .iter()
-                .map(|authority_id| authority_id.to_short_string())
-                .collect::<Vec<_>>()
-                .join(", ");
-            write!(
-                f,
-                "incoming only - {:?} [{}]; ",
-                self.incoming_peers.len(),
-                peers
-            )?;
-        }
-
-        if !self.outgoing_peers.is_empty() {
-            let peers = self
-                .outgoing_peers
-                .iter()
-                .map(|authority_id| authority_id.to_short_string())
-                .collect::<Vec<_>>()
-                .join(", ");
-            write!(
-                f,
-                "outgoing only - {:?} [{}];",
-                self.outgoing_peers.len(),
-                peers
-            )?;
+        match wanted_incoming {
+            0 => write!(f, "not expecting any incoming connections; ")?,
+            _ => {
+                write!(f, "expecting {:?} incoming connections; ", wanted_incoming)?;
+                match self.incoming_peers.is_empty() {
+                    true => write!(f, "WARNING! No incoming peers even though we expected tham, maybe connecting to us is impossible; ")?,
+                    false => write!(
+                            f,
+                            "have - {:?} [{}]; ",
+                            self.incoming_peers.len(),
+                            pretty_authority_id_set(&self.incoming_peers),
+                    )?,
+                }
+                if !self.missing_incoming.is_empty() {
+                    write!(
+                        f,
+                        "missing - {:?} [{}]; ",
+                        self.missing_incoming.len(),
+                        pretty_authority_id_set(&self.missing_incoming),
+                    )?;
+                }
+            }
         }
 
-        if !self.missing_peers.is_empty() {
-            let peers = self
-                .missing_peers
-                .iter()
-                .map(|authority_id| authority_id.to_short_string())
-                .collect::<Vec<_>>()
-                .join(", ");
-            write!(f, "missing - {:?} [{}];", self.missing_peers.len(), peers)?;
+        match wanted_outgoing {
+            0 => write!(f, "not attempting any outgoing connections; ")?,
+            _ => {
+                write!(f, "attempting {:?} outgoing connections; ", wanted_outgoing)?;
+                if !self.outgoing_peers.is_empty() {
```

Comment:
aren't we concerned if we cannot connect to any node? (similarly to the lack of incoming connections)

Reply:
The warning about incoming connections is there only because we expect this to be a common misconfiguration and it's _much_, **much**, I cannot stress _how_ much (seriously, I tried imagining it and it would probably require actually trying **hard**) harder to misconfigure network in a way that allows incoming connections, but breaks outgoing.

Added a comment as to why we warn about this and not the other way around, because you are probably right it's not obvious what our reasoning is _a priori_.